RFCs/FS-1329-Allow typed bindings in CE let!s without parentheses.md

```diff
@@ -0,0 +1,283 @@
+- [x] [Discussion](https://github.com/fsharp/fslang-design/discussions/XXXX)
+
+# Summary
+
+Allow computation expression (CE) bindings type annotations without requiring parentheses:
+* `let!` `and!` to accept type annotations on simple patterns without requiring parentheses, making them consistent with regular `let` bindings. 
+* `use!` to accept type annotations on simple patterns without requiring parentheses, making them consistent with regular `use` bindings.
+
+# Motivation
+
+Currently, when using type annotations with `let!`, `use!`, and `and!` bindings in computation expressions, parentheses are always required around the pattern with its type annotation, even for simple identifiers. This differs from regular `let` and `use` bindings where parentheses are only required for complex patterns.
+
+This inconsistency is:
+- **A syntactical foot-gun** that is frustrating and confusing to users
+- **Inconsistent** with the general design of F# where `let!` should behave similarly to `let` where possible
+- **A common source of compiler errors** for developers working with computation expressions
+- **Particularly problematic for F# newcomers** who are already grappling with computation expressions
+- **Unhelpful error messages**: Getting "Unexpected symbol ':'" doesn't immediately suggest that parentheses are needed
+
+# Detailed design
+
+## Scenario: `let!` bindings
+
+`let!` bindings with type annotations on simple identifiers require parentheses:
+
+```fsharp
+// Before
+async {
+    let! (data: byte[]) = readAsync()
+    let! (x: int) = async { return 42 }
+}
+
+// After  
+async {
+    let! data: byte[] = readAsync()
+    let! x: int = async { return 42 }
+}
+```
+
+## Scenario: `use!` bindings
+
+`use!` bindings with type annotations on simple identifiers require parentheses:
+
+```fsharp
+// Before
+task {
+    use! (conn: SqlConnection) = openConnectionAsync()
+    use! (resource: IDisposable) = openResourceAsync()
+}
+
+// After
+task {
+    use! conn: SqlConnection = openConnectionAsync()
+    use! resource: IDisposable = openResourceAsync()
+}
+```
+
+## Scenario: `and!` bindings
+
+`and!` bindings with type annotations on simple identifiers require parentheses:
+
+```fsharp
+// Before
+async {
+    let! (userId: int) = getUserIdAsync()
+    and! (permissions: Permission list) = getPermissionsAsync()
+    and! (profile: UserProfile) = getProfileAsync()
+}
+
+// After
+async {
+    let! userId: int = getUserIdAsync()
+    and! permissions: Permission list = getPermissionsAsync()
+    and! profile: UserProfile = getProfileAsync()
+}
+```
+
+## Scenario: Tuple patterns
+
+Tuple patterns with type annotations require parentheses (same as regular `let`):
+
+```fsharp
+// Before (and still required after)
+async {
+    let! (a, b): int * string = asyncTuple()
+    and! (x, y): float * bool = asyncPair()
+}
+
+// After - parentheses still required (consistent with regular let)
```

Comment:
How about tuples with each element having direct type annotation?

`let (x:int,y:int) = (5,3)` equivalent

I assume the same as regular let as well?

Reply:
Yes. this wont be allowed. Will update it mention this.